site/src/AppRouter.tsx

```diff
@@ -315,6 +333,15 @@ export const AppRouter: FC = () => {
                   path="external-auth"
                   element={<ExternalAuthSettingsPage />}
                 />
+
+                <Route path="oauth2-provider">
+                  <Route path="apps">
```

Comment:
One thing I noticed during testing: I think there should be a redirect or something when the user tries to go go the `deployment/oauth2-provider/` URL path

Right now, it doesn't take you to a completely empty screen, but you only see the navbar and the Deployment options

Reply:
I am hesitant to do that because one day we might have something else on that page (like for example a toggle for whether to enable the oauth2 provider) and I would not want folks getting confused learning they can go to just `/oauth2-provider` to get to the apps, if that makes sense.

Reply:
That said, seems like it should at least show a proper 404.  Mine gives this error: 
```
"user" must be an existing uuid or username.
queried user="deployment"
```
So I guess it is falling through to some other page.

Reply:
Whoops I was on `dev.coder.com`, yeah you are right it shows a blank page with the deployment sidebar.  That is pretty weird.  Is there even a way for me to show a 404 here?  Maybe a redirect is the only fix...

Reply:
I added a route with `NotFoundPage`.

Reply:
Ah but that is weird too because the sidebar is still there.  At least the page says "not found" I guess.